controlplane/pkg/model/client_connection_editor_manager.go

```diff
@@ -0,0 +1,134 @@
+package model
+
+import (
+	"fmt"
+)
+
+// ClientConnectionEditor is a type to commit changes on ClientConnection
+type ClientConnectionEditor struct {
+	ClientConnection *ClientConnection
+
+	id              string
+	connectionState ClientConnectionState
+}
+
+type clientConnectionEditorManager struct {
+	domain  clientConnectionDomain
+	editors map[string]*ClientConnectionEditor
+
+	allowedTransitions map[ClientConnectionState]map[ClientConnectionState]bool
+}
+
+func newClientConnectionEditorManager() clientConnectionEditorManager {
+	return clientConnectionEditorManager{
+		domain:  newClientConnectionDomain(),
+		editors: map[string]*ClientConnectionEditor{},
+
+		allowedTransitions: map[ClientConnectionState]map[ClientConnectionState]bool{
+			ClientConnectionReady: {
+				ClientConnectionRequesting: true,
+				ClientConnectionHealing:    true,
+				ClientConnectionClosing:    true,
+			},
+			ClientConnectionRequesting: {
+				ClientConnectionReady:   true,
+				ClientConnectionBroken:  true,
+				ClientConnectionClosing: true,
+			},
+			ClientConnectionBroken: {
+				ClientConnectionHealing: true,
+				ClientConnectionClosing: true,
+			},
+			ClientConnectionHealing: {
+				ClientConnectionRequesting: true,
+				ClientConnectionClosing:    true,
+			},
+			ClientConnectionClosing: {
+			},
+		},
+	}
+}
+
+func (m *clientConnectionEditorManager) AddClientConnection(id string, connectionState ClientConnectionState, cc *ClientConnection) (*ClientConnectionEditor, error) {
```

Comment:
Why `connectionState ClientConnectionState` is separate parameter? I think it could be set as field of `cc`